winsup/cygwin/dcrt0.cc

```diff
@@ -1360,7 +1360,10 @@ multiple_cygwin_problem (const char *what, uintptr_t magic_version, uintptr_t ve
     system_printf ("%s magic number mismatch detected - %p/%ly", what, magic_version, version);
   else
     api_fatal ("%s mismatch detected - %ly/%ly.\n\
-This problem is probably due to using incompatible versions of the cygwin DLL.\n\
+This problem is probably due to using incompatible versions of the msys2 or cygwin DLL.\n\
+msys2 is uses pinned memory to share child information. All versions running must be the same magic_version.\n\
```

Comment:
It would probably be a good idea to provide more verbose documentation about this somewhere on https://msys2.org/, and to add a link to that documentation to this here error message.

Reply:
Done! Other PR is pending here 

https://github.com/msys2/msys2.github.io/pull/235